src/pro/main/ipc/handlers/visual_editing_handlers.ts

```diff
@@ -0,0 +1,386 @@
+import { ipcMain } from "electron";
+import fs from "node:fs";
+import { promises as fsPromises } from "node:fs";
+import path from "path";
+import { db } from "../../../../db";
+import { apps } from "../../../../db/schema";
+import { eq } from "drizzle-orm";
+import { getDyadAppPath } from "../../../../paths/paths";
+import { parse } from "@babel/parser";
+import * as recast from "recast";
+import traverse from "@babel/traverse";
+import {
+  stylesToTailwind,
+  extractClassPrefixes,
+} from "../../../../utils/style-utils";
+import git from "isomorphic-git";
+import { gitCommit } from "../../../../ipc/utils/git_utils";
+
+interface StyleChange {
+  componentId: string;
+  componentName: string;
+  relativePath: string;
+  lineNumber: number;
+  appId: number;
+  styles: {
+    margin?: { left?: string; right?: string; top?: string; bottom?: string };
+    padding?: { left?: string; right?: string; top?: string; bottom?: string };
+    dimensions?: { width?: string; height?: string };
+    border?: { width?: string; radius?: string; color?: string };
+    backgroundColor?: string;
+    text?: {
+      fontSize?: string;
+      fontWeight?: string;
+      color?: string;
+      fontFamily?: string;
+    };
+  };
+  textContent?: string;
+}
+
+export function registerVisualEditingHandlers() {
+  ipcMain.handle(
+    "apply-visual-editing-changes",
+    async (_event, changes: StyleChange[]) => {
+      try {
+        if (changes.length === 0) return;
+
+        // Get the app to find its path (all changes should be for the same app)
```

Comment:
this is a little weird, you should instead define appId at the top-level if it's really going to be the same across all changes.